scheduler/filter/filter.go

```diff
@@ -14,6 +15,9 @@ type Filter interface {
 
 	// Return a subset of nodes that were accepted by the filtering policy.
 	Filter(*cluster.ContainerConfig, []*node.Node, bool) ([]*node.Node, error)
+
+	// Return a list of constraints/filters provided
+	GetAllFilters(*cluster.ContainerConfig) ([]string, error)
```

Comment:
I feel `All` is not necessary. Another reason is you are calling `GetAllFilters` in `ApplyFilters` and `listAllFilters`. For me it reduces readability.